tests/query/bug2457-001.phpt

```diff
@@ -0,0 +1,34 @@
+--TEST--
+PHPC-2457: Query modifiers can be passed reference
+--FILE--
+<?php
+
+$modifiers = ['$orderby' => ['x' => 1]];
```

Comment:
You noted the following error in [PHPC-2457](https://jira.mongodb.org/browse/PHPC-2457):

> PHP Fatal error: Uncaught MongoDB\Driver\Exception\InvalidArgumentException: Expected "$orderby" modifier to be array or object, bool given

The exception originates from [`php_phongo_query_opts_append_document()`](https://github.com/mongodb/mongo-php-driver/blob/1.20.0/src/MongoDB/Query.c#L62), but I'm not sure how that's possible as the code calls `PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(value)`. That macro delegates to [`zend_get_type_by_const()`](https://github.com/php/php-src/blob/php-8.3.12/Zend/zend_API.c#L110). `IS_REFERENCE` isn't even included in that function, and I can't imagine why the type would match for the function returning "bool".